src/Mapbender/CoreBundle/Component/Cache/Backend/File.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+
+namespace Mapbender\CoreBundle\Component\Cache\Backend;
+use Mapbender\BaseKernel;
+use Mapbender\CoreBundle\Component\Exception\CacheMiss;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Cache backend using the file system.
+ * This uses the default Symfony cache directories also used by twig, doctrine etc as the base storage location.
+ *
+ * Cache backend provides raw key:string value storage and handles automatic invalidation based on "marks"
+ * (which can be timestamps, but can also be more complex, as long as they are serializable).
+ *
+ * Instance registered in container at mapbender.presenter.application.cache.backend (see serviccs.xml)
+ */
+class File
+{
+    /** @var string */
+    protected $rootPath;
+
+    public function __construct(ContainerInterface $container)
+    {
+        /** @var BaseKernel $kernel */
+        $kernel = $container->get('kernel');
+        $rootPath = realpath($kernel->getCacheDir());
+        if (!is_dir($rootPath) || !is_writable($rootPath)) {
+            throw new \RuntimeException("Cache path {$rootPath} is not writable");
+        }
+        $this->rootPath = $rootPath;
+    }
+
+    /**
+     * @param string[] $keyPath a multi-component key, in this backend used as a subdirectory hierarchy
+     * @param string $value will be stored exactly as given
+     * @param mixed $mark to determine reusability of the cache entry later in get
+     */
+    public function put($keyPath, $value, $mark)
+    {
+        // prepend root path
+        array_unshift($keyPath, $this->rootPath);
+        $fullPath = implode('/', $keyPath);
+        @mkdir(dirname($fullPath));
```

Comment:
Are you sure it's a good idea to use @? I always have a bad feeling regarding things like this...

Reply:
It's just a cache. None of its contents are required for the application to function. Everything that goes into the cache can be generated again.
As such, I'm ignoring interaction errors in put silently so the application can keep running.

Likewise, errors from file_put_contents are ignored as well.

The worst possible end result of errors in either mkdir or file_put_contents is a cache miss when the system looks for reusable data. Cache misses are also not error conditions.